.github/workflows/build.yml

```diff
@@ -5,24 +5,59 @@ on:
     pull_request:
 
 jobs:
-    build:
-        runs-on: ubuntu-latest
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout project
+        uses: actions/checkout@v4
 
-        steps:
-          - uses: actions/checkout@v3
-          - uses: actions/setup-java@v3
-            with:
-              java-version: '17'
-              distribution: 'zulu'
-          - uses: gradle/gradle-build-action@v2
-          - name: Build and run unit tests with Gradle
-            run: ./scripts/ci_unit.sh
-          - name: Build javadocs
-            run: ./gradlew :dokkaHtmlMultiModule
-          - name: Publish a source snapshot to Sonatype
-            env:
-              NEXUS_USERNAME: ${{ secrets.NEXUS_USERNAME }}
-              NEXUS_PASSWORD: ${{ secrets.NEXUS_PASSWORD }}
-            if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' && github.repository == 'bumptech/glide' }}
-            run: ./gradlew publish -PNEXUS_USERNAME="${NEXUS_USERNAME}" -PNEXUS_PASSWORD="${NEXUS_PASSWORD}"
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: "zulu"
+          java-version: "17"
 
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Validate Gradle wrapper
+        uses: gradle/actions/wrapper-validation@v4
```

Comment:
setup-gradle does this since v4:
https://github.com/gradle/actions/releases/tag/v4.0.0